### ISSUE TITLE
Implement variant endings more fully

### ISSUE THREAD

**Issue by newlandsvalley** [newlandsvalley/purescript-abc-parser#27]
At the moment, the parser supports variant endings of the form ```|n <notes>``` but the midi module ignores these for values of n above 2.  i.e. only 2 variant endings are supported and this is also true in ```abc-melody```

The parser does not support variants of the form ```1,3```  or ```1-3```.  Note, too, that the ```P:``` - parts header is marked as volatile in the spec and thus subject to change.
,
It seems to me that we ought to extend the variant endings support to the Midi module (and to abc-melody) whilst still ignoring the Parts setting for the time being.  This is probably the most significant and awkward improvement we could make to abc-parser. 

**4.10 Variant endings**
```
In combination with P: part notation, it is possible to notate more than two variant endings for a section that is to be repeated a number of times.

For example, if the header of the tune contains P:A4.B4 then parts A and B will each be played 4 times. To play a different ending each time, you could write in the tune:

P:A
<notes> | [1  <notes>  :| [2 <notes> :| [3 <notes> :| [4 <notes> |]
The Nth ending starts with [N and ends with one of ||, :| |] or [|. You can also mark a section as being used for more than one ending e.g.

[1,3 <notes> :|
plays on the 1st and 3rd endings and

[1-3 <notes> :|
plays on endings 1, 2 and 3. In general, '[' can be followed by any list of numbers and ranges as long as it contains no spaces e.g.

[1,3,5-7  <notes>  :| [2,4,8 <notes> :|
```

**Comment by newlandsvalley** [comment]
The voltas branch now supports the ```!1,3``` form.  The ```|1-3``` form seems much less widespread.

**Comment by newlandsvalley** [comment]
I think I know how best to add volta ranges of the form ```|1-3```.  Voltas will become a non-empty list of either the simple form or the volta range form.  These will be macro-expanded to a simple list of volta numbers before being handed to the Midi module which needs to handle the repeats.  i.e. ```|1,2-5....``` becomes effectively ```|1,2,3,4,5 ```.

**Comment by newlandsvalley** [comment]
Variant ranges (as well as all the others) now fixed by 4f34c31c85a96909c8783136e4b68bd4abb32477
